src/components/Organisms/index.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { useRef, useState, useEffect } from "react";
+import styled from "styled-components";
+import FONTFAMILY from "../../variables/font_family";
+import TEXTS from "../../variables/texts";
+import COLOR from "../../variables/color";
+import AddTaskButton from "../Atoms/AddTaskButton/index";
+import Tasks from "../Molecules/index";
+
+const TodoCard = () => {
+  const [taskList, setTaskList] = useState([{ name: "", initializing: true }]);
+
+  const onAddTaskButtonClick = () => {
+    setTaskList([...taskList,({ name: "", initializing: true })]);
+  };
+
+  const onTaskComplete = ({ index }) => {
+    const deleteList = [...taskList];
+    //console.log(index);
+    deleteList.splice(index, 1);
+    setTaskList(deleteList);
+    //setTaskList([[...taskList].slice(0, index), [...taskList].slice(index+1)])
+  };
+
+  const onTaskNameChange = ({ value, index }) => {
+    const changeList = [...taskList];
+    //console.log(value+" "+index);
+    if (value) {
+      //changeList[index].name = value;
+      changeList.splice(index, 1,{ name: value, initializing: true })
+    } else {
+      changeList.splice(index, 1);
+    }
+    setTaskList(changeList);
+  };
+
+  return (
+    <StyledWrapper>
+      <AddTaskButton onClick={onAddTaskButtonClick} />
+      <StyledTaskList>
+        {taskList.map((task, index) => {
+          return (
+            <Tasks
+              key={index}
+              onTaskChange={(value) => {
+                onTaskNameChange({ value, index });
+              }}
+              onTaskComplete={(index) => {
+                onTaskComplete(index);
```

Comment:
```suggestion
                onTaskComplete({index});
```
こんな感じに修正するとよいかと！

解説：
16行目の関数定義の部分で、 `{index: 何らかの値 }` を受け取るようになっていますが、現状 `index` (number) を第一引数にそのまま渡してしまっているため、 `onTaskComplete` 実行時には  `index` が `undefined` になってしまっています。

これにより、「どこをクリックしても一番上の要素が消える」という状況が再現しているものと思われます。

![image](https://github.com/JunPlayGround/Jun-frontend-basic-2023-todoapp/assets/83479175/2b4f1c67-d471-405c-99b2-32d7cb72b411)